packages/web3-validator/src/utils.ts

```diff
@@ -187,7 +187,16 @@ export const abiSchemaToJsonSchema = (
 				delete childSchema.minItems;
 			}
 
-			lastSchema.items = childSchema;
+			// lastSchema.items is a Schema, concat with 'childSchema'
+			if (!Array.isArray(lastSchema.items)) {
+				lastSchema.items = [lastSchema.items as JsonSchema, childSchema];
+			} // lastSchema.items is an empty Scheme array, set it to 'childSchema'
+			else if (lastSchema.items.length === 0) {
+				lastSchema.items = childSchema;
+			} // lastSchema.items is a non-empty Scheme array, append 'childSchema'
+			else {
+				lastSchema.items.push(childSchema);
```

Comment:
need a unit test hitting this `lastSchema.items.push(childSchema);`

Reply:
Will do that in a follow-up PR (with another fix that covers another use-case for the implementation)

Reply:
Thanks @fullkomnun